pydantic/networks.py

```diff
@@ -189,7 +190,8 @@ def validate(cls, value: Any, field: 'ModelField', config: 'BaseConfig') -> 'Any
             raise errors.UrlSchemePermittedError(cls.allowed_schemes)
 
         user = parts['user']
-        if cls.user_required and user is None:
+        password = parts['password']
+        if cls.user_required and user is None and password is None:
```

Comment:
Humm, this is a breaking change (and arguably a bug) since now user is no longer required when `user_required = True` - just password will suffice .

To fix this, either we need to:
1. add `password_required` (and then probably `user_or_password_required`, maybe `credentials_required` would be a better name)
2. or, leave this logic unchanged and just change `user_required = False` on `RedisDsn`

I think I favour option 1., but on `RedisDsn` it looks like neither user or password are required, so it doesn't make much difference in this case.